tests/unit/test_table.py

```diff
@@ -5393,6 +5409,270 @@ def test_set_expiration_w_none(self):
         assert time_partitioning._properties["expirationMs"] is None
 
 
+class TestPrimaryKey(unittest.TestCase):
+    @staticmethod
+    def _get_target_class():
+        from google.cloud.bigquery.table import PrimaryKey
+
+        return PrimaryKey
+
+    @classmethod
+    def _make_one(cls, *args, **kwargs):
+        return cls._get_target_class()(*args, **kwargs)
+
+    def test_constructor_explicit(self):
+        columns = ["id", "product_id"]
+        primary_key = self._make_one(columns)
+
+        self.assertEqual(primary_key.columns, columns)
+
+    def test__eq__columns_mismatch(self):
+        primary_key = self._make_one(columns=["id", "product_id"])
+        other_primary_key = self._make_one(columns=["id"])
+
+        self.assertNotEqual(primary_key, other_primary_key)
+
+    def test__eq__other_type(self):
+        primary_key = self._make_one(columns=["id", "product_id"])
+
+        self.assertNotEqual(primary_key, object())
```

Comment:
TIL:

`isinstance(X, object)`

always returns `True` because of a call to `type.__instancecheck__()`.

Essentially, this is because all elements in Python ARE `object`s (and not just DERIVED from `object`s).

So this test, which uses `isinstance()` behind the scenes will always fail if we test `.assertNotEqual()` using `object`).